### PR TITLE
Pubsub testing

### PR DESCRIPTION
**Pull request by EricBolten**
A happy-path pubsub integration test.

### CODE REVIEW

File: integration_tests/pubsub_test.go [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+		s.Require().NoError(err, "Unable to create governance proposal")
+
+		s.submitAndVoteForProposal(proposerCtx, orch0ClientCtx, propID, removeDefaultSubscriptionPropMsg)
+		propID++
+
+		s.T().Log("Verifying DefaultSubscription correctly removed")
+		defaultSubscriptionsResponse, err = pubsubQueryClient.QueryDefaultSubscriptions(context.Background(), &types.QueryDefaultSubscriptionsRequest{})
+		s.Require().NoError(err)
+		s.Require().Len(defaultSubscriptionsResponse.DefaultSubscriptions, 0)
+
```
Comment (cbrit) [inline integration_tests/pubsub_test.go:311]: I think we're missing removal of the publisher intent? Which raises the question, should the RemovePublisher handler delete the corresponding publisher intents? Same question for subscribers.

**Review by cbrit — COMMENTED** [review]
One change needed

File: integration_tests/pubsub_test.go (hunk 1 above)
Comment (EricBolten) [reply integration_tests/pubsub_test.go:311]: Ah yes you're right, I forgot to remove the PublisherIntent.

I believe the code is supposed to delete Intents when either a Publisher or Subscriber is deleted, and also remove SubscriberIntents when a PublisherIntent is deleted, but I hadn't included that in this test yet. It's a good idea to do.

**Review by cbrit — APPROVED** [review]
LGTM

**Pull request merged**